crates/nu-command/src/filters/group_by.rs

```diff
@@ -238,16 +238,64 @@ pub fn group_by(
         return Ok(Value::record(Record::new(), head).into_pipeline_data());
     }
 
-    let mut groupers = groupers.into_iter();
+    let mut closure_idx = 0;
+
+    // Error early on unsupported types by collecting
+    let groupers = groupers
+        .into_iter()
+        .map(|val| match val {
+            Value::CellPath { val, .. } => Ok(Grouper::CellPath { val }),
+            Value::Closure {
+                val, internal_span, ..
```

Comment:
Typically, we don't allow internal_span to be used since the plan is to eventually remove them. I think the way we solved this previously, iirc, is to get the span for each grouper and use it. I could be saying that wrong.

Reply:
Removing it caused a bit of a refactor but it's honestly so much better now.
The structure is clearer and it also led to more granular spans for the error messages. I'll push the commit(s) shortly.

Thanks for the heads up.